src/ucs/stats/stats.c

```diff
@@ -112,6 +117,76 @@ static void ucs_stats_clean_node(ucs_stats_node_t *node) {
     ucs_list_del(&node->type_list);
 }
 
+static void ucs_stats_free_class(ucs_stats_class_t *cls)
+{
+    unsigned i;
+
+    for (i = 0; i < cls->num_counters; i++) {
+        ucs_free((void*)cls->counter_names[i]);
+    }
+
+    ucs_free((void*)cls->name);
+    ucs_free(cls);
+}
+
+static ucs_stats_class_t *ucs_stats_dup_class(ucs_stats_class_t *cls)
+{
+    ucs_stats_class_t *dup;
+
+    dup = ucs_calloc(1, sizeof(*cls) + sizeof(*cls->counter_names) * cls->num_counters,
+                     "ucs_stats_class_dup");
+    if (!dup) {
+        ucs_warn("failed to allocate statistics class");
+        goto err;
+    }
+
+    dup->name = ucs_strdup(cls->name, "ucs_stats_class_t name");
+    if (!dup->name) {
+        ucs_warn("failed to copy statistics class name");
+        goto err_free;
+    }
+
+    for (dup->num_counters = 0; dup->num_counters < cls->num_counters; dup->num_counters++) {
+        dup->counter_names[dup->num_counters] = ucs_strdup(cls->counter_names[dup->num_counters],
+                                                           "ucs_stats_class_t counter");
+        if (!dup->counter_names[dup->num_counters]) {
+            ucs_warn("failed to copy statistics counter name");
```

Comment:
ucs_error("failed to allocate statistics counter name");

Reply:
fixed